pep-0458.txt

```diff
@@ -509,11 +509,43 @@ discarded and more offline keys will not help resist key recovery attacks [21]_
 unless diversity of keys is maintained.
 
 
-Online and Offline Keys Recommended for Each Role
--------------------------------------------------
+Managing online keys
+---------------------
 
-In order to support continuous delivery, the *timestamp*, *snapshot*, *bins*
-role keys MUST be online.
+In order to support continuous delivery, the keys for the *timestamp*,
+*snapshot*, all *bin-n* roles MUST be online. There is little benefit in
+requiring all of these roles to use different online keys, since attackers
+would presumably be able to compromise all of them if they compromise PyPI.
+Therefore, it is reasonable to use one online key for them all.
+
+This online key MAY be stored, encrypted or not, on the Python infrastructure.
+For example, the kept MAY be kept on a self-hosted key management service
+(e.g. Hashicorp Vault__), or a third-party one (e.g. AWS KMS__, Google
+Cloud KMS__, or Azure Key Vault__).
+
+__ https://www.vaultproject.io/
+__ https://aws.amazon.com/kms/
+__ https://cloud.google.com/kms/
+__ https://docs.microsoft.com/en-us/azure/key-vault/basic-concepts
+
+Some of these key management services allow keys to be stored on Hardware
+Security Modules (HSMs) (e.g., Hashicorp Vault__, AWS CloudHSM__, Google
+Cloud HSM__, Azure Key Vault__). This prevents attackers from exfiltrating
+the online private key (although not from using it, although their actions
```

Comment:
Minor nit: Maybe replace one of the two 'although's?

Reply:
Yes, I think I fixed this later, thanks :)